### PR TITLE
feat(defaults) :enable customization of path_display.truncate

### PR DESCRIPTION
**Pull request by tami5**
followup to #1254 and implementation of[@Conni2461's idea](https://github.com/nvim-telescope/telescope.nvim/pull/1254#issuecomment-922527270)

- [x] Fix the empty of empty lines.
- [x] edit docs
- [x] fix lint

### CODE REVIEW

**Comment by tami5** [comment]
does make sense to make default to 2?

Screen Shot 2021-09-19 at 11 50 39 PM

File: lua/telescope/utils.lua [hunk 1]
```diff
@@ -372,6 +372,11 @@ utils.transform_path = function(opts, path)
       if vim.tbl_contains(path_display, "truncate") or path_display.truncate then
         if opts.__length == nil then
           opts.__length = calc_result_length()
+          if type(path_display.truncate) == "number" then
```
Comment (Conni2461) [inline lua/telescope/utils.lua:375]: can we actually do the calculation in the `calc_result_length` local function?

File: lua/telescope/utils.lua [hunk 2]
```diff
@@ -372,6 +372,11 @@ utils.transform_path = function(opts, path)
...
+            opts.__length = opts.__length - path_display.truncate
+          else
+            opts.__length = opts.__length - 2
```
Comment (Conni2461) [inline lua/telescope/utils.lua:378]: I think we dont need a default padding

**Review by l-kershaw — COMMENTED** [review]
I agree with @Conni2461's points, and don't have anything further to suggest.

**Pull request merged**